local.go

```diff
@@ -351,16 +367,22 @@ func (drive LocalDrive) listFolder(path string, prefix string, config *ListConfi
 
 	return res, nil
 }
-
-func (drive LocalDrive) checkName(path string) (string, error) {
-	_, err := os.Stat(path)
-
-	for !os.IsNotExist(err) {
-		path = path + ".new"
-		_, err = os.Stat(path)
-	}
-
-	return path, nil
+func (drive LocalDrive) checkName(path string, mode string) (string, error) {
+    _, err := os.Stat(path)
+
+    for !os.IsNotExist(err) {
+        
+        if mode == "folder" {
+            path = path + ".new"
+        } else {
+            index := strings.LastIndex(path, ".")
```

Comment:
How it will work for file names without extension ? ( some/Dockerfile )
Probably using https://golang.org/pkg/path/filepath/#Ext will help

Reply:
it will, thanks)

Reply:
fixed